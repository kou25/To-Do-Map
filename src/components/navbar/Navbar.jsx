import React from 'react'
import note from './note.png'
import ToggleButton from'../Sidebar/ToggleButton'
 const Navbar = props => (
        <header className="nav-head">
            <nav className="navigation">
            <div><ToggleButton click={props.clickHandler}/></div>
            <div className="nav-logos">
                <a data-testid="img" href='/' style={{textShadow:"1px 1px 2px black"}}>
                    <img  src={note} alt="logo" style={{height:"70px"}}/>
                    TO DO
                </a>
            </div>
            <div className="spacer"/>
            </nav>
        </header>
)


export default Navbar;