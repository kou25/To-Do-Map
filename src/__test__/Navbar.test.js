import React from 'react';
import ReactDom from 'react-dom'
import Navbar from '../components/navbar/Navbar'
import {render, cleanup} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'

afterEach(cleanup);
it("render without crashing",()=>{
  const div= document.createElement("div");
  ReactDom.render(<Navbar/>,div);
})

it("renders links correctly", ()=>{
  const{getByTestId}=render(
  <Navbar/>)
  expect(getByTestId('img')).toHaveTextContent('TO DO')
})

it('matches snapshot', ()=>{
  const tree=renderer.create(
  <Navbar/>).toJSON()
  expect(tree).toMatchSnapshot()
})