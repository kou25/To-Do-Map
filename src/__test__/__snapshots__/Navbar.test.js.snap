// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="nav-head"
>
  <nav
    className="navigation"
  >
    <div>
      <button
        className="button"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </button>
    </div>
    <div
      className="nav-logos"
    >
      <a
        data-testid="img"
        href="/landing"
        style={
          Object {
            "textShadow": "1px 1px 2px black",
          }
        }
      >
        <img
          alt="logo"
          src="note.png"
          style={
            Object {
              "height": "70px",
            }
          }
        />
        TO DO
      </a>
    </div>
    <div
      className="spacer"
    />
  </nav>
</header>
`;
